rest/server_context.go

```diff
@@ -839,14 +868,74 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(ctx context.Context, config
 		return dbcontext, nil
 	}
 
-	atomic.StoreUint32(&dbcontext.State, db.DBOnline)
-	_ = dbcontext.EventMgr.RaiseDBStateChangeEvent(dbName, "online", stateChangeMsg, &sc.Config.API.AdminInterface)
+	// If asyncOnline wasn't specified, block until db init is completed, then start online processes
+	if !options.asyncOnline {
+		base.DebugfCtx(ctx, base.KeyConfig, "Waiting for database init to complete...")
+		if dbInitDoneChan != nil {
```

Comment:
I don't know if you can hit this pathway, but do you need to make sure we aren't using persistent config here?

Reply:
Do you mean clarify that we *are* using persistent config?  You might be right that an additional check is needed, just trying to work out which combination you're thinking of.